test/SimulationCSharpClient.Tests/HttpRetryHandler/HttpRetryMessageHandlerTests.cs

```diff
@@ -54,5 +54,63 @@ public void HttpGet_Success_In_FirstTry()
             // Assert
             Assert.AreEqual(((TestHttpRetryMessageHandlerSuccess)handler.InnerHandler).AttemptedTries, 1);
         }
+
+        [Test]
+        public void HttpRetryMessageHandler_ThrowsException_If_Cancellation_Requested()
+        {
+            // Arrange
+            int maxReTries = 3;
+            var cancellationTokenSource = new CancellationTokenSource();
+            var httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, "http://localhostTestApi.com");
+
+            var handler = new HttpRetryMessageHandler(maxReTries)
+            {
+                InnerHandler = new TestHttpRetryMessageHandlerFailure()
+            };
+
+            var invoker = new HttpMessageInvoker(handler);
+
+            // Act
+            cancellationTokenSource.Cancel();
+            var exception = Assert.Catch<Exception>(() =>
+            {
+                var result = invoker.SendAsync(httpRequestMessage, cancellationTokenSource.Token).Result;
+            });
+
+            // Assert
+            Assert.IsNotNull(exception);
+            Assert.IsNotNull(exception is OperationCanceledException);
+            Assert.AreEqual(((TestHttpRetryMessageHandlerFailure)handler.InnerHandler).AttemptedTries, 0);
+        }
+
+        [Test]
+        public void HttpRetryMessageHandler_WaitAndRetryAsync_ThrowsException_If_Cancellation_Requested()
+        {
+            // Arrange
+            int maxReTries = 3;
+            var cancellationTokenSource = new CancellationTokenSource();
+            var httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, "http://localhostTestApi.com");
+
+            var handler = new HttpRetryMessageHandler(maxReTries)
+            {
+                InnerHandler = new TestHttpMessageHandlerCancelAfterFirstTry()
+                {
+                    CancellationTokenSource = cancellationTokenSource
+                }
+            };
+
+            var invoker = new HttpMessageInvoker(handler);
+
+            // Act
+            var exception = Assert.Catch<Exception>(() =>
```

Comment:
do you want to test for a specific exception type?

Reply:
Yes, but I am expecting either OperationCancelledException OR TaskCancelledException (which wrapped in Aggregate Exception ) so Assert.Throws<OperationCancelledException> was not working.

Reply:
the aggregate exception would only come from the http, not from the ThrowIf calls right?

Reply:
We are throwing on retry in HttpHandlers so we are getting aggregate exception also